app/src/main/java/com/kickstarter/viewmodels/ViewPledgeViewModel.java

```diff
@@ -248,8 +255,18 @@ public ViewModel(final @NonNull Environment environment) {
         .compose(bindToLifecycle())
         .subscribe(this.shippingSectionIsGone);
 
-      this.viewMessagesButtonIsGone = this.currentConfig.observable()
-        .map(config -> !ObjectUtils.coalesce(config.features().get(FeatureKey.ANDROID_MESSAGES), false));
+      Observable.zip(
+        isFromMessagesActivity,
+        featureFlagIsEnabled,
+        Pair::create
+      )
+        .map(fromMessagesAndFlagEnabled -> fromMessagesAndFlagEnabled.first || !fromMessagesAndFlagEnabled.second)
+        .compose(bindToLifecycle())
+        .subscribe(this.viewMessagesButtonIsGone::onNext);
```

Comment:
ah yes you have identified a weirdness/subtlety with our outputs that we're still trying to figure out-- 

our `isFromMessagesActivity` is a boolean that comes from our `intent()`, and we need the output to be a `BehaviorSubject` because for some reason an `Observable` doesn't emit immediately, due to our base `intent()` being a `PublishSubject`

tl;dr whenever you have an output that relies on data coming from the `intent()` it needs to be a `BehaviorSubject`